types/time.go

```diff
@@ -1020,6 +1022,15 @@ func parseDatetime(sc *stmtctx.StatementContext, str string, fsp int8, isFloat b
 
 			year, month, day, hour, minute, second =
 				dateTime.Year(), dateTime.Month(), dateTime.Day(), dateTime.Hour(), dateTime.Minute(), dateTime.Second()
+			// Only for case: 0.XXX
+			if seps[0] == "0" {
+				microsecond, overflow, err = ParseFrac(fracStr, fsp)
+				if err != nil {
+					return ZeroDatetime, errors.Trace(err)
+				}
+				truncatedOrIncorrect = overflow
```

Comment:
This line looks useless?  This branch is never use `truncatedOrIncorrect`. 

By the way, I think this logic is same as following
```
if seps[0] == "0" || l >= 9 && l <= 14 {
  hhmmss = true
}
```
I had tried it, and passed all the tests.

Reply:
It's not added by me. commit info as follows:
```
commit 8ed9a6ff6d8bb5d9c74f936d01182d4859ab14d1 
Author: Lenshood Date: 2020/10/15, 3:46 下午 
 expression: fix TIMESTAMP func get wrong result with decimal (#15185) (#20088)
```

Reply:
I mean that the logic of this code
```
switch len(seps) {
case 1:
	if isFloat {
		// do something
		if seps[0] == "0" {
			// do something
			microsecond, overflow, err = ParseFrac(fracStr, fsp)
			if err != nil {
				return ZeroDatetime, errors.Trace(err)
			}
			truncatedOrIncorrect = overflow
		}
		// do something
		if l >= 9 && l <= 14 {
			hhmmss = true
		}
		break
	}
}
// do something
if hhmmss {
	microsecond, overflow, err = ParseFrac(fracStr, fsp)
	if err != nil {
		return ZeroDatetime, errors.Trace(err)
	}
}
```
 is the same as 
```
switch len(seps) {
case 1:
	if isFloat {
		// do something
		if seps[0] == "0" ||l >= 9 && l <= 14 {
			hhmmss = true
		}
		break
	}
}
// do something
if hhmmss {
	microsecond, overflow, err = ParseFrac(fracStr, fsp)
	if err != nil {
		return ZeroDatetime, errors.Trace(err)
	}
}
```

Reply:
Yep, maybe the semantic is a little different.